apps/teams/tasks.py

```diff
@@ -1,10 +1,26 @@
 from celery import shared_task
 
 from apps.teams.invitations import send_invitation_accepted
-from apps.teams.models import Invitation
+from apps.teams.models import Invitation, Team
+from apps.utils.deletion import (
+    delete_object_with_auditing_of_related_objects,
+    get_admin_emails_with_delete_permission,
+    send_team_deleted_notification,
+)
 
 
 @shared_task(ignore_result=True)
 def send_invitation_accepted_notification(invitation_id):
     invitation = Invitation.objects.get(id=invitation_id)
     send_invitation_accepted(invitation)
+
+
+@shared_task
+def delete_team_async(team_id):
+    team = Team.objects.get(id=team_id)
+
+    # get required info for emails prior to deleting team info
+    team_name = team.name
+    admin_emails = get_admin_emails_with_delete_permission(team)
```

Comment:
Should all users maybe be notified instead of only those that can delete the team? It might be a bit surprising for someone to find their team is gone. Probably an edge case, but still

Reply:
FYI there is a limit of 50 recipients per email so this will need to be chunked. See https://github.com/dimagi/open-chat-studio/pull/1202